pkg/asset/agent/agentconfig/agent_config_test.go

```diff
@@ -0,0 +1,278 @@
+								Model:              "model-value",
+								Vendor:             "vendor-value",
+								SerialNumber:       "serial-number-value",
+								MinSizeGigabytes:   20,
+								WWN:                "wwn-value",
+								WWNWithExtension:   "wwn-with-extension-value",
+								WWNVendorExtension: "wwn-vendor-extension-value",
+								Rotational:         falsePtr,
+							},
+							Interfaces: []*aiv1beta1.Interface{
+								{
+									Name:       "enp2s0",
+									MacAddress: "98:af:65:a5:8d:01",
+								},
+								{
+									Name:       "enp3s1",
+									MacAddress: "28:d2:44:d2:b2:1a",
+								},
+							},
+						},
+						{
+							Hostname: "control-1.example.org",
+							Role:     "master",
+							Interfaces: []*aiv1beta1.Interface{
+								{
+									Name:       "enp2s0",
+									MacAddress: "98:af:65:a5:8d:02",
+								},
+								{
+									Name:       "enp3s1",
+									MacAddress: "28:d2:44:d2:b2:1b",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name:          "not-yaml",
+			data:          `This is not a yaml file`,
+			expectedError: "failed to unmarshal agent-config.yaml: error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type agent.Config",
+		},
+		{
+			name:       "file-not-found",
+			fetchError: &os.PathError{Err: os.ErrNotExist},
+		},
+		{
+			name:          "error-fetching-file",
+			fetchError:    errors.New("fetch failed"),
+			expectedError: "failed to load agent-config.yaml file: fetch failed",
+		},
+		{
+			name: "unknown-field",
+			data: `
+		metadata:
+		  name: agent-config-wrong
+		spec:
+		  wrongField: wrongValue`,
+			expectedError: "failed to unmarshal agent-config.yaml: error converting YAML to JSON: yaml: line 2: found character that cannot start any token",
+		},
+		{
+			name: "interface-missing-mac-address-error",
+			data: `
+metadata:
+  name: agent-config-cluster0
+spec:
+  rendezvousIP: 192.168.111.80
+  hosts:
+    - hostname: control-0.example.org
+      interfaces:
+        - name: enp2s0
+        - name: enp3s1
+          macAddress: 28:d2:44:d2:b2:1a`,
+			expectedError: "invalid Agent Config configuration: Spec.Hosts.Interfaces.macAddress: Required value: each interface must have a MAC address defined",
+		},
+		{
+			name: "node-hostname-and-role-are-not-required",
+			data: `
+metadata:
+  name: agent-config-cluster0
+spec:
+  rendezvousIP: 192.168.111.80
+  hosts:
+    - interfaces:
+        - name: enp3s1
+          macAddress: 28:d2:44:d2:b2:1a`,
+			expectedFound: true,
+			expectedConfig: &agent.Config{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "agent-config-cluster0",
+				},
+				Spec: agent.Spec{
+					RendezvousIP: "192.168.111.80",
+					Hosts: []agent.Host{
+						{
+							Interfaces: []*aiv1beta1.Interface{
+								{
+									Name:       "enp3s1",
+									MacAddress: "28:d2:44:d2:b2:1a",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Not a blocking point for this specific PR, but I guess in future it would be useful to adopt a fluent interface approach for writing the cases under the tests, since especially for larger / more complex objects it could be difficult to spot/read the relevant parts